### PR TITLE
[5.2] Blade `or` behaviour

### PR DESCRIPTION
**Pull request by KennedyTedesco**
This is a proposal that brings to `or` operator the same ideia (but not the behavior, of course) of the `Null Coalesce Operator`.

The ideia of `or` in Blade is to **echo** a variable if exists, otherwise a default value is printed. And, of course, all of this using a ternary statement. So, instead of:

```
{{ isset($name) ? $name : 'Default' }}
```

We just write:

```
{{ $name or 'Default' }}
```

Cool!

But, knowing the ideia of `or` is to be a short-cut to **print** data, **maybe** is a good thing it also verify if the value is **not null**. Something like:

```
{{ isset($name) && !is_null($name) ? $name : 'Default' }}
```

If this change is a breaking change? I don't think so. Who needs to print a null value?

What you guys think?

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
I think this was already proposed and rejected was it not?

**Comment by GrahamCampbell** [comment]
Or, just write...

```
{{ $name ?? 'Default' }}
```

**Comment by KennedyTedesco** [comment]
I do not remember. But if was closed before: :+1:

Yeah, that works fine for PHP7 people. And to be honest `or` is useless on PHP7 (since we have `??`).

But on my project with PHP 5.6 this is a good thing. I'll extend Blade. :)

File: tests/View/ViewBladeCompilerTest.php [hunk 1]
```diff
@@ -92,7 +92,7 @@ public function testEchosAreCompiled()
         $this->assertEquals('<?php echo $name; ?>', $compiler->compileString('{!!
             $name
         !!}'));
-        $this->assertEquals('<?php echo isset($name) ? $name : \'foo\'; ?>', $compiler->compileString('{!! $name or \'foo\' !!}'));
+        $this->assertEquals('<?php echo isset($name) && !is_null($name) ? $name : \'foo\'; ?>', $compiler->compileString('{!! $name or \'foo\' !!}'));
```
Comment (cryode) [inline tests/View/ViewBladeCompilerTest.php]: FYI this is redundant. `isset()` already checks if the variable is null.

**Comment by cryode** [comment]
I tested options for this back in #10135. At the end of the day, it's not worth trying to support every possible edge case for the `or` syntax. Ideally I'd like to see it deprecated and suggest using standard PHP syntaxes like short ternary or null coalesce.

File: tests/View/ViewBladeCompilerTest.php (hunk 1 above)
Comment (KennedyTedesco) [inline tests/View/ViewBladeCompilerTest.php]: :+1: My mind wasn't clear. :smile:

**Comment by KennedyTedesco** [comment]
@cryode Yeah, this gonna happen naturally I guess.
